src/cryptography/x509/extensions.py

```diff
@@ -676,9 +683,8 @@ def __ne__(self, other):
 class NoticeReference(object):
     def __init__(self, organization, notice_numbers):
         self._organization = organization
-        if not isinstance(notice_numbers, list) or not all(
-            isinstance(x, int) for x in notice_numbers
-        ):
+        notice_numbers = list(notice_numbers)
+        if not all(isinstance(x, int) for x in notice_numbers):
```

Comment:
Behavior change: NoticeReference now allows notice_numbers to be any iterable, not just list